### PR TITLE
migrate obsm varm tests, skip if known issue

### PR DESCRIPTION
**Pull request by rcannood**
* Migrate obsm varm tests to use dummy_anndata in a similar way to #193
* Keep track of known issues
* Skip test if something is a known issue

dummy_anndata should also be able to generate data frames in obsm / varm. See data-intuitive/dummy-anndata#11

### CODE REVIEW

**Review by lazappi — APPROVED** [review]
I think this should be go. I guess it's nicer, as long as the file is kept up to date.

File: tests/testthat/test-roundtrip-X.R [hunk 1]
```diff
@@ -1,109 +1,104 @@
...
+library(reticulate)
+testthat::skip_if_not(
+  reticulate::py_module_available("dummy_anndata"),
+  message = "Python dummy_anndata module not available for testing"
+)
 
-test_names <- names(data$layers)
+ad <- reticulate::import("anndata", convert = FALSE)
+da <- reticulate::import("dummy_anndata", convert = FALSE)
+bi <- reticulate::import_builtins()
```
[the remark this answers is not in the document]
Comment (rcannood) [reply tests/testthat/test-roundtrip-X.R:12]: Is `ad` not the standard naming convention for importing anndata -- that is,

```python
import anndata as ad
```

Similarly, I thought it'd be funny if `dummy_anndata` would be imported as

```python
import dummy_anndata as da
```

I can change the builtins, though!

[the remark this answers is not in the document]
Comment (lazappi) [reply tests/testthat/test-roundtrip-X.R:12]: Oh, I guess so, I hadn't thought of that. It's only obvious if you know the convention though and TBH I'm not a fan of doing it in Python either, I usually import things with the full name unless it's something super common like **numpy**. It's a reasonable thing to do though so fine to keep it.

File: tests/testthat/test-roundtrip-obsvar.R [hunk 2]
```diff
@@ -39,62 +38,78 @@ for (name in test_names) {
   adata_py$write_h5ad(file_py)
 
   test_that(paste0("reading an AnnData with obs and var '", name, "' works"), {
+    msg <- message_if_known(
+      backend = "HDF5AnnData",
+      slot = c("obsp", "varp"),
```
[the remark this answers is not in the document]
Comment (rcannood) [reply tests/testthat/test-roundtrip-obsvar.R:43]: Yes!

There was a bit of copy pasting. I'm considering merging the tests into one. WDYT?

----

```suggestion
    msg <- message_if_known(
      backend = "HDF5AnnData",
      slot = c("obs", "var"),
```

**Pull request merged**